### PR TITLE
HDDS-7644. S3 multipart upload does not update quota namespace for missing parents

### PR DESCRIPTION
**Pull request by sumitagrawl**
## What changes were proposed in this pull request?
S3 multipart upload for FSO type creates a directory if missing for a parent. But the corresponding namespace addition is not recorded.
Validation and namespace quota updated for missing parent directory that got created.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7644

## How was this patch tested?

Tested using unit test for namespace quota

### CODE REVIEW

**Comment by kerneltime** [comment]
cc @neils-dev @duongkame can you'll please take a look.

**Review by sadanand48 — COMMENTED** [review]
Thanks @sumitagrawl  for the patch, Overall looks good, left a minor comment.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/response/s3/multipart/TestS3MultipartResponse.java [hunk 1]
```diff
@@ -326,7 +330,7 @@ public S3MultipartUploadCompleteResponse createS3CompleteMPUResponseFSO(
 
   protected S3InitiateMultipartUploadResponse getS3InitiateMultipartUploadResp(
       OmMultipartKeyInfo multipartKeyInfo, OmKeyInfo omKeyInfo,
-      OMResponse omResponse, long volumeId, long bucketId) {
+      OMResponse omResponse, long volumeId, long bucketId) throws IOException {
```
[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/response/s3/multipart/TestS3MultipartResponse.java:333]: This is added as derived class need throw exception, where bucketInfo is obtained.

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/response/s3/multipart/TestS3MultipartUploadAbortResponse.java [hunk 2]
```diff
@@ -100,7 +100,8 @@ public void testAddDBToBatch() throws Exception {
   protected S3InitiateMultipartUploadResponse
         getS3InitiateMultipartUploadResponse(
       String volumeName, String bucketName, String keyName,
-      String multipartUploadID, long volumeId, long bucketId) {
+      String multipartUploadID, long volumeId, long bucketId)
+      throws IOException {
```
[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/response/s3/multipart/TestS3MultipartUploadAbortResponse.java:104]: one method called inside that now throws IOException, so added to the signature.

**Review by sadanand48 — APPROVED** [review]
LGTM. +1

**Pull request merged**